File: backend/pkg/model/group.go
package model

type Group struct {
	Id        int    `json:"group_id" db:"id"`
	Name      string `json:"name" db:"name"`
	IsArchive bool   `json:"is_archive" db:"is_archive"`
}

type GroupResponse struct {
	Groups []Group `json:"groups"`
}

type GroupAndDisciplineInput struct {
	GroupId      int `json:"group_id" binding:"required"`
	DisciplineId int `json:"discipline_id" binding:"required"`
}

type GroupChangeNameInput struct {
	GroupId int    `json:"group_id" binding:"required"`
	Name    string `json:"name" binding:"required"`
}

type GroupId struct {
	Id int `json:"group_id" db:"group_id"`
}

File: frontend/src/models/group/group.ts
import { ISeminarian } from '../seminarian/seminarian';

export type IGroup = IGroup.Group;

export namespace IGroup {
  export interface Group {
    group_id: string;
    is_archive: boolean;
    name: string;
  }

  export interface GroupsFetchData {
    groups: IGroup.Group[];
  }

  export interface AddGroup {
    group_name: string;
  }

  export interface ChangeGroupName {
    name: string;
    group_id: number;
  }

  export interface DeleteGroup {
    id: string;
  }

  export interface FetchGroupStudents {
    id: string;
  }

  export interface GroupStudent {
    name: string;
    surname: string;
    user_id: string;
  }

  export interface GetGroupStudentStudentId {
    students: GroupStudentStudentId[];
  }

  export interface GroupStudentStudentId {
    name: string;
    surname: string;
    student_id: string;
  }

  export interface FetchGroupSeminarians {
    discipline_id: number;
    group_id: number;
  }

  export interface GroupSeminarians {
    seminarians: ISeminarian.Seminarian[];
  }

  export interface GroupStudents {
    students: GroupStudent[];
  }

  export interface AddSeminarian {
    discipline_id: number;
    group_id: number;
    seminarian_id: number;
  }
}
